app/controllers/profile_controller.rb

```diff
@@ -3,4 +3,27 @@ def show
     render file: "/public/404" if current_user.nil?
```

Comment:
I must have missed this earlier, but we should be calling `visitor?` instead of `current_user.nil?`